src/app/app.component.ts

```diff
@@ -162,4 +165,20 @@ export class AppComponent implements OnInit {
     return false;
   }
 
+  /**
+   * retrieve stack config information: api endpoint url, and then store it into
+   * localStorage + inject to every request (RequestModule)
+   *
+   * @param   {string}  stackUuid  uuid in string
+   * @return  {void}
+   */
+  retrieveStackConfig(stackUuid: string): void {
+    if (stackUuid) {
+      this.authService.getStackConfig(stackUuid).subscribe(res => {
+        this.storage.stackConfig = res;
+      });
+    }
+
+    return this.storage.stackConfig;
```

Comment:
we don't need to return it here. If we just return it here, it doesn't have the value from the API